src/queue.c

```diff
@@ -6264,8 +6264,12 @@ _dispatch_worker_thread(void *context)
 static unsigned WINAPI
 _dispatch_worker_thread_thunk(LPVOID lpParameter)
 {
-  _dispatch_worker_thread(lpParameter);
-  return 0;
+	HRESULT hr = CoInitializeEx(NULL, COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE);
+	if (SUCCEEDED(hr)) {
+		_dispatch_worker_thread(lpParameter);
+		CoUninitialize();
+	}
```

Comment:
This would mean that failure to initialize COM would terminate the thread immediately - even if COM is never used on the thread.  Are we certain that there are no cases in which COM initialization could fail that would not be recoverable?  (e.g. if COM initialization fails, we know that either you have no memory)

Reply:
I hoisted the succeeded check based on your previous comment.

I am not sure which way is the right way to do this (ignore COM initialization failure or not). I have a hard time finding good documentation on COM initialization that would answer your questions, and I can see arguments for both options. That being said, I’m leaning to how it is implemented now as I am guessing COM initialization is pretty unlikely to fail (other than out of memory) as it’s such an integral part of Windows programming.

Reply:
I'm worried about the diagnosability of that.  Perhaps it makes sense to make the failure fatal then?  Right now, if this fails, we don't have any notification but are unable to recover from that because any attempt to create a thread will fail.

Reply:
given the return values specified here: https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-coinitializeex#return-value

i think a `fatalError` is OK since we know we aren't going to pass invalid arguments, and OOM or E_UNEXPECTED sound like fine reasons to `fatalError` to me

Reply:
Sounds good. What’s the best way to trigger a fatal error here? I think `fatalError()` is only available in Swift. Would `_dispatch_assert_crash()` make sense, or any other function/macro from internal.h?

Reply:
yeah `_dispatch_assert_crash` would be the best way.

Reply:
Thanks. I ended up using `_dispatch_client_assert_fail` because it takes a format string, which enables adding the error code to the message.